src/PowerShell/Microsoft.WinGet.Client.Engine/PSObjects/PSInstallResult.cs

```diff
@@ -16,14 +15,50 @@ namespace Microsoft.WinGet.Client.Engine.PSObjects
     public sealed class PSInstallResult
     {
         private readonly InstallResult installResult;
+        private readonly CatalogPackage? catalogPackage;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="PSInstallResult"/> class.
         /// </summary>
         /// <param name="installResult">The install result COM object.</param>
-        internal PSInstallResult(InstallResult installResult)
+        /// <param name="catalogPackage">The catalog package COM object.</param>
+        internal PSInstallResult(InstallResult installResult, CatalogPackage? catalogPackage)
         {
             this.installResult = installResult;
+            this.catalogPackage = catalogPackage;
+        }
+
+        /// <summary>
+        /// Gets the id of the installed package.
+        /// </summary>
+        public string Id
+        {
+            get
+            {
+                return this.catalogPackage != null ? this.catalogPackage.Id : string.Empty;

```

Comment:
For my own knowledge, Is there a reason to not use null coalescing for these?

Reply:
I forgot about those 😓 , updated to use null coalescing operators